vendor/src/github.com/docker/engine-api/types/types.go

```diff
@@ -274,6 +274,19 @@ type ExecStartCheck struct {
 	Tty bool
 }
 
+type HealthcheckResult struct {
+	Start, End time.Time
+	ExitCode   int    // 0=healthy, 1=unhealthy, 2=starting
+	Output     string // Output from last check
```

Comment:
It's limited to 255 bytes, so images would be tricky. I've updated the docs to specify UTF-8: https://github.com/talex5/docker/blob/healthcheck/docs/reference/builder.md#healthcheck

> To help debug failing probes, any output text (UTF-8 encoded) that the command writes on stdout or stderr will be stored in the health status and can be queried with docker inspect. Such output should be kept short (only the first 255 bytes are stored currently).